### ISSUE TITLE
Optimize type-scoped contexts

### ISSUE THREAD

**Issue by brechtvdv** [rubensworks/jsonld-context-parser.js#34]
JSON-LD contexts that contain type-specific context blocks currently have a high parsing complexity.

For example when parsing Linked Art's context:
```
jsonld-context-parse url https://brechtvdv.github.io/demo-data/linkedart.jsonld
```
This takes (infinite) time to parse.

This can be optimized in future releases of this tool.

**Comment by rubensworks** [comment]
Thanks for reporting!

For future reference: this is likely due to the large number of type-scoped contexts, which seem to lead to an exponential increase in execution time.

**Comment by azaroth42** [comment]
Hi both.
I can try and write a minimal context that has only the "official" predicates and classes from linked art.  We started off this way and then as we moved properties and classes in or out of the profile, we found it easier to just have it with the complete set, lest we miss something and the triples silently disappear.  Let me know if that would be useful and I'll add it to the to-do list :)

**Comment by rubensworks** [comment]
@azaroth42 Thanks! That may be useful indeed.
I'm not sure yet if it's really a problem with the context though.

It looks like this parser's execution time is exponential in terms of the number of type-scoped context entries.
This may either be an inherent property of the JSON-LD parsing algorithm, or just an implementation issue on our end, not sure yet.

**Comment by rubensworks** [comment]
Fixed in 2.1.2!
